### PR TITLE
ENH: NDArrayOperatorsMixin calls ufuncs directly, like ndarray

### PR DESCRIPTION
**Pull request by shoyer**
Per our discussion in https://github.com/numpy/numpy/pull/8247#discussion_r112825050

### CODE REVIEW

File: numpy/lib/mixins.py [hunk 1]
```diff
@@ -5,53 +5,54 @@
 
 from numpy.core import umath as um
 
-# None of this module should be exposed in top-level NumPy module.
+# Nothing should be exposed in the top-level NumPy module.
 __all__ = []
 
 
+def _disables_array_ufunc(obj):
```
Comment (mhvk) [inline numpy/lib/mixins.py]: I think it makes more sense to put this function on the class itself (and thus use `self._disables_array_ufunc` in the function definitions below). That way we define the default, but do not force people to reimplement the mixin if they do not like the default.

Comment (shoyer) [inline numpy/lib/mixins.py]: Can you please explain the use-case you are envisioning in a little more detail? To be clear, you want to make it possible to subclass the mixin and override this check? For what end?

I am skeptical that making it easier for users to shoot themselves in the foot is a good thing :).

Comment (charris) [inline numpy/lib/mixins.py]: @shoyer I tend to agree with your position, OTOH, putting it in the class is easy to do and puts everything in one bundle. The fact that it is private should serve as warning to users that the warranty is void if they choose to fool with it.

Comment (shoyer) [inline numpy/lib/mixins.py]: I think this helper function is already logically linked with the class due to it's location in this module.

Comment (charris) [inline numpy/lib/mixins.py]: OK. Easy enough to change if experience warrants it.

**Comment by charris** [comment]
Thanks Stephan.

**Pull request merged**